File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/rippleeffect/data/RippleEffectDocument.java
package com.zwstudio.logicpuzzlesandroid.puzzles.rippleeffect.data;

import com.zwstudio.logicpuzzlesandroid.common.data.GameDocument;
import com.zwstudio.logicpuzzlesandroid.common.data.GameProgress;
import com.zwstudio.logicpuzzlesandroid.common.data.MoveProgress;
import com.zwstudio.logicpuzzlesandroid.common.domain.Position;
import com.zwstudio.logicpuzzlesandroid.puzzles.rippleeffect.domain.RippleEffectGame;
import com.zwstudio.logicpuzzlesandroid.puzzles.rippleeffect.domain.RippleEffectGameMove;
import com.zwstudio.logicpuzzlesandroid.puzzles.rippleeffect.domain.RippleEffectObject;

import org.androidannotations.annotations.EBean;

/**
 * Created by zwvista on 2016/09/29.
 */

@EBean
public class RippleEffectDocument extends GameDocument<RippleEffectGame, RippleEffectGameMove> {
    protected void saveMove(RippleEffectGameMove move, MoveProgress rec) {
        rec.row = move.p.row;
        rec.col = move.p.col;
        rec.intValue1 = move.obj.ordinal();
    }
    public RippleEffectGameMove loadMove(MoveProgress rec) {
        return new RippleEffectGameMove() {{
            p = new Position(rec.row, rec.col);
            obj = RippleEffectObject.values()[rec.intValue1];
        }};
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/rippleeffect/RippleEffectDocument.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.rippleeffect

import android.content.Context
import com.zwstudio.logicpuzzlesandroid.common.data.GameDocument
import com.zwstudio.logicpuzzlesandroid.common.data.MoveProgress
import com.zwstudio.logicpuzzlesandroid.common.domain.Position

class RippleEffectDocument(context: Context) : GameDocument<RippleEffectGameMove>(context) {
    override fun saveMove(move: RippleEffectGameMove, rec: MoveProgress) {
        rec.row = move.p.row
        rec.col = move.p.col
        rec.intValue1 = move.obj
    }

    override fun loadMove(rec: MoveProgress) =
        RippleEffectGameMove(Position(rec.row, rec.col), rec.intValue1)
}